starlette/_utils.py

```diff
@@ -0,0 +1,12 @@
+import asyncio
+import functools
+import typing
+
+
+def iscoroutinefunction(obj: typing.Any) -> bool:
```

Comment:
This is a nit, but should we call this `is_async_callable`?

AFAICT, having "something we can call and `await`" is all we care about when checking for apps or endpoints.

"Coroutine function" is a well-defined word in the Python glossary: a "function which returns a coroutine object", defined with "`async def`".

https://docs.python.org/3/glossary.html#term-coroutine-function

Also, the current naming makes it look as a compatibility shim on `asyncio.iscoroutinefunction` which focuses on checking just the definition above^, whereas we do some more checks, such as looking for `__call__`.

Reply:
> This is a nit, but should we call this is_async_callable?

Sure. 👍 I'll adapt later today. Thanks for the review. 🙏